pkg/tfvars/internal/cache/cache.go

```diff
@@ -226,3 +232,110 @@ func DownloadImageFile(baseURL string) (string, error) {
 
 	return DownloadFile(baseURL, imageDataType)
 }
+
+func detectContentType(file string) (string, error) {
+	logrus.Debugf("Detecting content type for %s", file)
+	fd, err := os.Open(file)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to open file %s: %v", file, err)
+		return "", err
+	}
+	defer fd.Close()
+
+	// See http://golang.org/pkg/net/http/#DetectContentType for why we use 512
+	buf := make([]byte, 512)
+	_, err = fd.Read(buf)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to read file %s: %v", file, err)
+		return "", err
+	}
+	filetype := http.DetectContentType(buf)
+	logrus.Debugf("detectContentType %s is %s", file, filetype)
+	return filetype, nil
+}
+
+func decompressFileGzip(src, dest string) error {
+	logrus.Debugf("decompressFileGzip %s %s", src, dest)
+	gzipfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	reader, err := gzip.NewReader(gzipfile)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	writer, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+	//defer os.Remove(dest)
+
+	if _, err = io.Copy(writer, reader); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func decompressFileXZ(src, dest string) error {
+	logrus.Debugf("decompressFileXZ %s %s", src, dest)
+	xzfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer xzfile.Close()
+
+	reader, err := xz.NewReader(xzfile)
```

Comment:
Tangentially see https://github.com/coreos/mantle/pull/1170/commits/c7b6994a2a74e43530933b36fffdc757932f5b72

Of course the mantle codebase has a lot of stuff around this area, including now signature checking...maybe at some point we could try to make it a shared library.